net/tailscale/files/tailscale.init

```diff
@@ -4,44 +4,266 @@
-  /usr/sbin/tailscaled --cleanup
+custom_instance() {
+  local cfg="$1"
+  local accept_routes hostname accept_dns advertise_exit_node exit_node advertise_routes site_to_site subnet_routes flags login_server authkey std_out std_err
+
+  config_get_bool accept_routes $cfg 'accept_routes'
+  config_get hostname $cfg 'hostname'
+  config_get_bool accept_dns $cfg 'accept_dns'
+  config_get_bool advertise_exit_node $cfg 'advertise_exit_node'
+  config_get exit_node $cfg 'exit_node'
+  config_get advertise_routes $cfg 'advertise_routes'
+  config_get_bool site_to_site $cfg 'site_to_site'
+  config_get flags $cfg 'flags'
+  config_get login_server $cfg 'login_server'
+  config_get authkey $cfg 'authkey'
+  config_get_bool std_out $cfg 'log_stdout'
+  config_get_bool std_err $cfg 'log_stderr'
 
   procd_open_instance
-  procd_set_param command /usr/sbin/tailscaled
+  procd_set_param command $PROG up --reset
 
-  # Starting with v1.48.1 ENV variable is required to enable use of iptables / nftables.
-  # Use nftables by default - can be changed to 'iptables' in tailscale config
-  procd_set_param env TS_DEBUG_FIREWALL_MODE="$fw_mode"
+  [ "$accept_routes" = "1" ] && procd_append_param command --accept-routes=true
+  [ -n "$hostname" ] && procd_append_param command --hostname="$hostname"
+  [ "$accept_dns" = "0" ] && procd_append_param command --accept-dns=false
+  [ "$advertise_exit_node" = "1" ] && procd_append_param command --advertise-exit-node
+  [ -n "$exit_node" ] && {
+    procd_append_param command --exit-node="$exit_node"
+    procd_append_param command --exit-node-allow-lan-access=true
+  }
+  [ -n "$advertise_routes" ] && procd_append_param command --advertise-routes="$(echo $advertise_routes | tr ' ' ',')"
+  [ "$site_to_site" = "1" ] && procd_append_param command --snat-subnet-routes=false
+  [ -n "$flags" ] && procd_append_param command $flags
+  [ -n "$login_server" ] && procd_append_param command --login-server="$login_server"
+  [ -n "$authkey" ] && procd_append_param command --authkey="$authkey"
+
+  procd_set_param stdout "$std_out"
+  procd_set_param stderr "$std_err"
+  procd_close_instance
+  (
+    # Check if the wait PID file exists, if so, return
+    [ -f "/var/run/tailscale.wait.pid" ] && return
+    # Create the wait PID file
+    touch /var/run/tailscale.wait.pid
+
+    # Wait for Tailscale IPv4 address to be available
+    count=0
+    while [ -z "$(ifconfig | grep 'tailscale' | awk '{print $1}')" ] || [ -z "$(tailscale ip -4)" ]
+    do
+      sleep 2
+      let count++
+      [ "${count}" -ge 5 ] && { rm /var/run/tailscale.wait.pid; exit 19; }
+    done
+
+    # Configure Tailscale MagicDNS if accept_dns is enabled
+    if [ "$accept_dns" = "1" ]; then
+      MagicDNSSuffix=$(tailscale status --json | awk -F'"' '/"MagicDNSSuffix"/ {last=$(NF-1)} END {print last}')
+      sed -i '/100.100.100.100/d' /etc/dnsmasq.conf
+      echo "server=/$MagicDNSSuffix/100.100.100.100" >> /etc/dnsmasq.conf
+      /etc/init.d/dnsmasq reload
```

Comment:
I think you should do something like:

```bash
uci add_list dhcp.cfg01411c.address='/$MagicDNSSuffix/100.100.100.100'
```

with a corresponding instance, instead of modifying a config file directly. And I guess this becomes more complicated when another DNS server is used or there are multiple instances of dnsmasq.

Also I think it's better to use the service helper instead of calling scripts directly, in case the path or the way the services are defined changes:

```bash
service dnsmasq reload
```

Reply:
You can look up all the existing instance of `dnsmasq` using

```bash
service dnsmasq info
```

Or do something like:

```bash
for i in $(seq 0 99); do
	if conf=$(uci show dhcp.@dnsmasq[$i]); then
		echo $i
	else
		break;
	fi
done
```

Reply:
You could do something like:

```bash
for i in $(seq 0 99); do
	conf=$(uci show dhcp.@dnsmasq[$i] 2>/dev/null)
	if [ $? -ne 0 ]; then
		break
	else
		uci add_list dhcp.@dnsmasq[$i].address="/$MagicDNSSuffix/100.100.100.100"
	fi
done
```

Reply:
What is the issues with the suggested approach? This seems to work just fine for deleting:

```bash
uci get dhcp.@dnsmasq[$i].address | sed -E "s|/$MagicDNSSuffix/[0-9.]+||"
```

And then update the entry.